src/app/account/update-password/update-password.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+
+import { APP_ROUTE_URLS, LOGIN_ERROR_MESSAGES } from '@constants/app-constants';
+import { AuthService } from 'app/shared/services/auth.service';
+
+@Component({
+  selector: 'app-update-password',
+  templateUrl: './update-password.component.html',
+  styleUrls: ['./update-password.component.scss']
+})
+export class UpdatePasswordComponent {
+
+  public passwordGroup = new FormGroup({
+    password: new FormControl('', [Validators.required, Validators.minLength(8)]),
+    confirmPassword: new FormControl('', [Validators.required, Validators.minLength(8)])
+  }, {
+      validators: this.isPasswordSame
+    });
+  public hidePassword = true;
+  public hideConfirmPassword = true;
+  public errorMessage: string;
+  public showError = false;
+
+  constructor(
+    private authService: AuthService,
+    private router: Router,
+  ) { }
+
+  isPasswordSame(formGroup: FormGroup) {
+    const password = formGroup.get('password').value;
+    const confirmPassword = formGroup.get('confirmPassword').value;
+    return password === confirmPassword ? null : { notSame: true };
+  }
+
+  get password() {
+    return this.passwordGroup.get('password');
+  }
+
+  onSubmit() {
+    const password = this.authService.encryptPassword(this.passwordGroup.get('password').value);
+    const updatePasswordData = {
+      'email': this.authService.getEmailAndReSendTime().email,
+      'otp': this.authService.getOTP(),
+      'password': password,
+    };
+    this.authService.updatePassword(updatePasswordData)
+      .subscribe(() => {
+        this.router.navigate([APP_ROUTE_URLS.login]);
+      },
+        (errorResponse: any) => {
+          if (errorResponse.status === 400) {
+            this.errorMessage = errorResponse.data.error;
+          } else {
+            this.errorMessage = LOGIN_ERROR_MESSAGES.internalError;
+          }
+          this.showError = true;
```

Comment:
Why do we need this? Is `errorMessage` not sufficient for the same?

Reply:
done.